cli/npm/global.d.ts

```diff
@@ -0,0 +1,44 @@
+type Rule =
+  | string
+  | RegExp
+  | {
+      /** @internal */
+      _ruleBrand: never;
+    };
+
+interface BaseGrammar<Keys extends string> {
+  name: string;
+
+  extras($: Record<Keys, Rule>): Rule[];
+
+  externals($: Record<Keys, Rule>): Rule[];
+
+  conflicts($: Record<Keys, Rule>): Rule[];
```

Comment:
According to https://tree-sitter.github.io/tree-sitter/creating-parsers#the-grammar-dsl, `conflicts` should return `Array<Array<Rule>>`.